### PR TITLE
HowTos Audit Logs

### PR DESCRIPTION
**Pull request by ilyakubanov**
Ticket: https://spryker.atlassian.net/browse/CC-33764

### CODE REVIEW

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+## Prerequisites
+
+Before you begin, ensure that the [Spryker Core feature](/docs/pbc/all/miscellaneous/{{page.version}}/install-and-upgrade/install-features/install-the-spryker-core-feature.html)
+is installed. Without it, you will not be able to proceed with the steps outlined in this guide.
+
+## Step-by-step instructions
+
+### Option 1: Introduce Processor Plugins
+
+* Obtain the **RequestStack** service to get access to the customer session:
```
Comment (AsonUnique) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md:25]: As for me, it would be more clear to rephrase it to something like 
"Add **RequestStack** service to the dependency provider of your module to get access to the customer session".

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+     */
+    protected function findCustomerInRequest(Request $request): ?CustomerTransfer
+    {
+        return $request->getSession()->get('customer data');
+    }
+}
+```
+
+* Register the newly introduced plugin for the **security** log type for the **Yves** application:
+*
```
Comment (AsonUnique) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md:184]: It seems like the asterisk is not needed here

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    }
+}
+```
+
+### Option 2: Pass the data to the specific log context
+
+Pass the data directly to the specific log context as needed:
+
+```php
+```php
```
Comment (AsonUnique) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md:215]: duplicated

**Review by AsonUnique — APPROVED** [review]

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+title: Extend the audit log structure
+description: Learn how to enhance your audit log structure.
+template: howto-guide-template
+---
+
+This document describes how to extend the audit log structure with additional data.
+As an example, a business unit of a logged-in customer for the `Yves` application will be added.
+
+There are two primary methods to extend data in audit logs:
+* Introduce processor plugins: if registered for the corresponding audit log type and application will always be executed and will try to extend the data.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md:11]: There seems to be an error in this sentence, double-check

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+        $container->set(static::SERVICE_REQUEST_STACK, function (Container $container) {
+            return $container->getApplicationService(static::SERVICE_REQUEST_STACK);
+        });
+
+        return $container;
+    }
+}
+```
+
+**src/Pyz/Yves/Log/LogFactory.php**
```
Comment (andriitserkovnyi) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md:75]: Instructions missing

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+    {
+        return $this->getProvidedDependency(LogDependencyProvider::SERVICE_REQUEST_STACK);
+    }
+}
+
+```
+
+* Introduce **CustomerBusinessUnitProcessorPlugin** which extracts the business unit name from the customer session and adds it to the log data.
+
+```php
```
Comment (andriitserkovnyi) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md:100]: What file is this?

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/add-audit-log-types.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+<?php
+
+use Pyz\Yves\Log\Plugin\Log\YvesCheckoutAuditLoggerConfigPlugin;
+use Pyz\Glue\Log\Plugin\Log\GlueBackendCheckoutAuditLoggerConfigPlugin;
+use Pyz\Glue\Log\Plugin\Log\GlueCheckoutAuditLoggerConfigPlugin;
+use Pyz\Zed\Log\Communication\Plugin\Log\MerchantPortalCheckoutAuditLoggerConfigPlugin;
+use Pyz\Zed\Log\Communication\Plugin\Log\ZedCheckoutAuditLoggerConfigPlugin;
+
+$config[LogConstants::AUDIT_LOGGER_CONFIG_PLUGINS_ZED] = [
+    //other plugins
```
Comment (profuel) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/add-audit-log-types.md:124]: does it mean that other plugins could be added only before the Core one?
Please mention this somewhere

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/add-audit-log-types.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+class AnyCheckoutClassWhereAuditLoggingIsNeeded
+{
+    use AuditLoggerTrait;
+
+    public function checkout()
+    {
+        //other logic
+
+        $this->getAuditLogger(
+            (new AuditLoggerConfigCriteriaTransfer())->setChannelName('checkout'),
```
Comment (profuel) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/add-audit-log-types.md:159]: @dunqan this sounds a bit too much for a simple operation selecting appropriate channel, especially, when there's also `tags` section. Could you please explain why?

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/add-audit-log-types.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+Now you can add audit logs with `Checkout` type across the application. Example:
+
+```php
+<?php
+
+use Generated\Shared\Transfer\AuditLoggerConfigCriteriaTransfer;
+use Spryker\Shared\Log\AuditLoggerTrait;
+
+class AnyCheckoutClassWhereAuditLoggingIsNeeded
```
Comment (profuel) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/add-audit-log-types.md:150]: what does this mean?
I'd prefer to demonstrate this on a real class.

@dunqan This is a clear violation of the principles we ask customers to follow - do not extend Core classes on the project without necessity. 
And here we actually explaining that this is the way to go forward...

File: docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+        ]);
+
+        //other logic
+    }
+}
+```
+
+## Summary
+
+Now when you log in as a customer with a business unit the corresponding business unit name will be added to the log data.
```
Comment (profuel) [inline docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md:248]: Quite artificial example, since providing a link to the customer in the log would've solved this task.

@dunqan overall feature seems to require a lot of customization, since the data storage is not connected to the environment - no cross-linking or similar. I understand, we can add links to the log, but that sounds wrong to me.

**Review by profuel — CHANGES_REQUESTED** [review]

File: docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+use Pyz\Yves\Log\Plugin\Log\YvesCheckoutAuditLoggerConfigPlugin;
+use Pyz\Glue\Log\Plugin\Log\GlueBackendCheckoutAuditLoggerConfigPlugin;
+use Pyz\Glue\Log\Plugin\Log\GlueCheckoutAuditLoggerConfigPlugin;
+use Pyz\Zed\Log\Communication\Plugin\Log\MerchantPortalCheckoutAuditLoggerConfigPlugin;
+use Pyz\Zed\Log\Communication\Plugin\Log\ZedCheckoutAuditLoggerConfigPlugin;
+
+$config[LogConstants::AUDIT_LOGGER_CONFIG_PLUGINS_ZED] = [
+    //other plugins
+    ZedCheckoutAuditLoggerConfigPlugin::class,
```
Comment (profuel) [inline docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:125]: does it mean that other plugins could be added only before the Core one?
Please mention this somewhere or add 
```
//other plugins
```
after this plugin.
In other places as well, please.

File: docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+class AnyCheckoutClassWhereAuditLoggingIsNeeded
+{
+    use AuditLoggerTrait;
+
+    public function checkout()
+    {
+        //other logic
+
+        $this->getAuditLogger(
+            (new AuditLoggerConfigCriteriaTransfer())->setChannelName('checkout'),
```
Comment (profuel) [inline docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:159]: @pyvovarov-s you said that this is optional, please mention it somewhere, so far all examples contain in-place transfer creation.
Also channel and tags usage is not clear here - when, why?

File: docs/pbc/all/miscellaneous/202407.0/extend-the-audit-log-structure.md [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+    protected function getYvesSecurityAuditLogProcessorPlugins(): array
+    {
+        return [
+            // other plugins
+            new CustomerBusinessUnitProcessorPlugin(),
+        ];
+    }
+}
+```
+Now when you log in as a customer with a business unit, the corresponding business unit name will be added to the log data.
```
Comment (profuel) [inline docs/pbc/all/miscellaneous/202407.0/extend-the-audit-log-structure.md:218]: I'd prefer to have a link to the customer, who did the action, so I can check there any details.
Did you think about this?

File: docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+    GlueBackendCheckoutAuditLoggerConfigPlugin::class,
+];
+$config[LogConstants::AUDIT_LOGGER_CONFIG_PLUGINS_MERCHANT_PORTAL] = [
+    //other plugins
+    MerchantPortalCheckoutAuditLoggerConfigPlugin::class,
+];
+```
+
+
+Now you can add audit logs with `Checkout` type across the application. Example:
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:142]: This is a good point, we need to refactor the example to showcase a safest way, which will help developers to always think about plugin way first

File: docs/pbc/all/miscellaneous/202407.0/extend-the-audit-log-structure.md (hunk 13 above)
Comment (ilyakubanov) [reply docs/pbc/all/miscellaneous/202407.0/extend-the-audit-log-structure.md:218]: The core functionality extends data with customer email/reference OOTB

File: docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md (hunk 12 above)
Comment (ilyakubanov) [reply docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:159]: AuditLoggerConfigCriteriaTransfer is not optional, in must be passed anyway.

Channel/tags are explained in another doc: https://github.com/spryker/spryker-docs/pull/2736 where all the information about audit logs is provided.

File: docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md (hunk 11 above)
Comment (ilyakubanov) [reply docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:126]: fixed

File: docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (ilyakubanov) [reply docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:147]: fixed

**Review by profuel — APPROVED** [review]

File: docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+];
+$config[LogConstants::AUDIT_LOGGER_CONFIG_PLUGINS_MERCHANT_PORTAL] = [
+    //other plugins
+    MerchantPortalCheckoutAuditLoggerConfigPlugin::class,
+    //other plugins
+];
+```
+
+
+Now you can add audit logs with `Checkout` type across the application. Example:
```
Comment (andriitserkovnyi) [inline docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:145]: How's this part related to the previous instructions?

Comment (ilyakubanov) [reply docs/pbc/all/miscellaneous/202407.0/add-audit-log-types.md:145]: We registered plugins for 'checkout' logs type so now it is possible to add them
